src/Illuminate/Database/Eloquent/Concerns/HasGeneratedColumns.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Illuminate\Database\Eloquent\Concerns;
+
+trait HasGeneratedColumns
+{
+    /**
+     * Perform any actions that are necessary after the model is saved.
+     *
+     * @param  array  $options
+     * @return void
+     */
+    protected function finishSave(array $options)
+    {
+        // We will reload the current model instance with fresh attributes from the
+        // database, making sure it has correct values for all generated columns
+        // after the model instance is saved just before firing "saved" event.
+        $this->refreshRawAttributes();
+
+        parent::finishSave($options);
```

Comment:
What would be the implications of doing this actually inside the `finishSave` method instead of building a trait?

Reply:
This actually runs a query to retrive and reset the attributes' values. It's not always needed but only when the model has generated columns.